### PR TITLE
Fix the behaviour of actionProductOutOfStock hook

### PR DESCRIPTION
**Pull request by loicm**
I opened 2 tickets:
- actually, actionProductOutOfStock was misnamed and should be named displayProductOutOfStock: http://forge.prestashop.com/browse/PSCSX-5928
- a real actionProductOutOfStock should exist: http://forge.prestashop.com/browse/PSCSX-5929

These are the 2 commits reflecting them.

### CODE REVIEW

**Comment by PrestaEdit** [comment]
I agree with you. The only things is that the hook actionProductOutOfStock is already use by some modules, like the mailalerts module.

Unfortunately, the name of the hook had to be good in the first time of apparition to got retro-compat compliant.

I think we need to rename the actionProductOutOfStock hook and set an alias about it.

But, with that, you need to have an another name for the real new (and correct) actionProductOutOfStock hook. :(

**Comment by PhpMadman** [comment]
Something simple as actionProductIsOutOfStock ?
Keeps the change to the minimum, and allows for the alias and retro compabilty

File: controllers/front/ProductController.php [hunk 1]
```diff
@@ -270,7 +270,7 @@ public function initContent()
 				'last_qties' =>  (int)Configuration::get('PS_LAST_QTIES'),
 				'HOOK_EXTRA_LEFT' => Hook::exec('displayLeftColumnProduct'),
 				'HOOK_EXTRA_RIGHT' => Hook::exec('displayRightColumnProduct'),
-				'HOOK_PRODUCT_OOS' => Hook::exec('actionProductOutOfStock', array('product' => $this->product)),
+				'HOOK_PRODUCT_OOS' => Hook::exec('displayProductOutOfStock', array('product' => $this->product)),
```
Comment (jeromenadaud) [inline controllers/front/ProductController.php]: Hi,

Thank you for your contribution, but as @PrestaEdit and @PhpMadman say, you can't change the name of the hook. but you can create a new one.

Best regards

**Comment by loicm** [comment]
If I sum-up:

- I create the new right/working hook and give it a new name. @PhpMadman suggests "actionProductIsOutOfStock". Other name suggestions or is that ok for you?
- As said by @jeromenadaud , I don't rename at the existing badly named hook "actionProductOutOfStock" to "displayProductOutOfStock" but I can add the alias "displayProductOutOfStock" instead. Alias is only adding an Hook::exec(), right?

**Comment by xBorderie** [comment]
(note: this is an automated message, but answering it will reach a real human )

Hi! Thank you for your pull request: your contribution to PrestaShop means a lot to us!

We are aware that this PR is still open after all this time, and we want to make sure that your hard work is not forgotten.

We are diving back into pull requests, and we want to work with you in order to get your PR to completion: we need your feedback!<br/>
Could you read [this article](http://build.prestashop.com/news/review-your-pull-requests/) from [the Build devblog](http://build.prestashop.com/), and take the time to review your pull requests, so that we can work with up-to-date information? You don't have to do it all right now, and every little step helps.

Again, thank you very much for this contribution, and for your renewed feedback on it!

-xb.

**Comment by xBorderie** [comment]
Without any feedback from the original author since the last automated request, and because the codebase as changed A LOT since the original commits, I am closing this PR. @loicm , don't hesitate to re-open it (and change the branch for 'develop') if you wish to rebase your code. Thank you for your contribution!

**Pull request closed without merging**